nio/service/base.py

```diff
@@ -263,9 +265,23 @@ def runproperties(self):
 
     def full_status(self):
         """Returns service plus block statuses for each block in the service"""
-        status = {"service": self.status.name}
-        for id in self._blocks:
-            status.update({id: self._blocks[id].status.name})
+
+        # build a general-like service status.
+        service_and_blocks_status = FlagsEnum(RunnerStatus)
+        # initialize it with the service status itself
+        service_and_blocks_status.flags = self.status.flags
+        # go trough its blocks and grab their warning and error statuses if any
+        for block_id in self._blocks:
+            if self._blocks[block_id].status.is_set(RunnerStatus.warning):
+                service_and_blocks_status.add(RunnerStatus.warning)
+            elif self._blocks[block_id].status.is_set(RunnerStatus.error):
+                service_and_blocks_status.add(RunnerStatus.error)
+
+        status = {"service": self.status.name,
```

Comment:
Since we're making changes here, we probably want these "special" status keys to be prefixed with underscores or something, to make them unique. As it stands, if you had a block called `"service"` or `"service_and_blocks"` (that would be weird...but whatever) they would collide. We either should nest the blocks in an inner object like this:
```
{
  "service": "started",
  "service_and_blocks": "warning",
  "blocks": {
    "MyBlock": "warning"
  }
}
```
Or replace `service` and `service_and_blocks` with `__service` and `__service_and_blocks` to indicate that they are special.

Reply:
Good point, I'll add the `blocks` nesting believing it will not affect the `system designer`